models/migrations/v158.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"code.gitea.io/gitea/modules/log"
+
+	"xorm.io/xorm"
+)
+
+func addUserPrimaryEmailToUserMails(x *xorm.Engine) error {
+	type User struct {
+		ID       int64  `xorm:"pk autoincr"`
+		Email    string `xorm:"NOT NULL"`
+		IsActive bool   `xorm:"INDEX"`
+	}
+	type EmailAddress struct {
+		ID          int64  `xorm:"pk autoincr"`
+		UID         int64  `xorm:"INDEX NOT NULL"`
+		Email       string `xorm:"UNIQUE NOT NULL"`
+		IsActivated bool
+	}
+
+	if err := x.Sync2(new(User), new(EmailAddress)); err != nil {
+		return err
+	}
```

Comment:
I don't think it's necessary, because you haven't changeed the struct of them.

Reply:
Got it, my bad, just read some other migrations and I thought it was required.